astropy/nddata/nduncertainty.py

```diff
@@ -156,26 +174,50 @@ def propagate_divide(self, other_nddata, result_data):
             The resulting uncertainty
         """
 
+        self._raise_not_implemented_arithmetic('Dividing')
+
+
+class CustomUncertainty(NDUncertainty):
+    """
+    A class for custom uncertainties
+
+    Parameters
+    ----------
+
+    array: ~np.ndarray
+
+    uncertainty_type: object
+        label to specify what kind of uncertainty you have (str is common)
```

Comment:
Looks like the attribute must exist and must be a string type (see https://github.com/astropy/astropy/blob/master/astropy/nddata/nddata_base.py#L87). IIRC, this was to ensure there would be a human-readable uncertainty type associated with the data.